### ISSUE TITLE
Bug: execute building will get an "Access is denied" error when vscode  is on

### ISSUE THREAD

**Issue by yymm120** [MystenLabs/sui#18749]
This seems to be a compatibility issue when `vscode` and `sui move build` are used together.  
All other rust projects, `cargo build or run` work well, even though vscode is still editing .rs.

**Reproduce step:**
1. `sui move new hello_world`
2. Open Vscode -> Open hello_world.move file -> click <kbd>ctrl</kbd> + <kbd>s</kbd> save it.
3. execute `sui move build` in external terminal.

**result:** 
- Expected:  build success.
- actual: I got a error:  (👇Yes, this is the error below. 👇)
```external-crates\move\crates\move-package\src\package_lock.rs "Access is denied"```

---
**Based on fixes provided by others**
Follow on repair strategies provided by others, I exclude the following folders from the security scan path.
```
- User/*****/.cargo
- User/*****/.move
- Path/to/my/sui/hello_world
```
<img src="https://github.com/user-attachments/assets/b361ebcd-71e9-44d4-aabd-09359f7ca30e" alt="image-20240720105848620" style="zoom: 70%;" />

**reference:**
1. https://github.com/rust-lang/cargo/issues/11544#issuecomment-1491691403
2. https://stackoverflow.com/questions/58909806/access-denied-executing-compiled-program/74170163#74170163

**after then, I got an another issue.** (_Same steps to reproduce._)
1. `sui move new hello_world`
2. Open Vscode -> Open hello_world.move file -> click <kbd>ctrl</kbd> + <kbd>s</kbd> save it.
3. execute `sui move build` in external terminal.

**result:** 
- Expected: build success.
- actual: I waited for 10 minutes and got **"nothing"**. (👇Yes, This is the second problem encountered.. 👇)
![image](https://github.com/user-attachments/assets/335e2edf-a845-42d9-afc3-64186e3fe0da)

---
os: windows 10
rust: 1.79.0
sui: 1.29.2

/(ㄒoㄒ)/~~
**If I close VScode it seems will work. But I obviously can't turn off vscode every time I build.** 

![image](https://github.com/user-attachments/assets/1fe6911d-03a0-4a4e-b8d7-d4f0bde1035a)

**Comment by yymm120** [comment]
Even though I know file will deny access if it opened.
but .ts .rs and .js will execute success or build success even though I opened it. I don't know tsc, rustc and node how to handle it. 

are there anyone encountered the same issue as me?

**Comment by awelc** [comment]
This is likely related to a different implementation of a cross-process lock on Windows vs. Mac/Linux that is used during package building - seems like on Windows the lock is not released properly (or in timely manner) at the end of the build process. We are looking into this but it's a rather tricky problem to solve...

**Comment by awelc** [comment]
I started looking into this. Could you help me a little (either @yymm120 or @parklife2023) and let me know what the Move analyzer log is showing (you get there by choosing `View-Output` from the VSCode menu and then selecting `Move` tab from the drop-down menu on the right-hand side)

**Comment by awelc** [comment]
Should be fixed in VSCode v1.0.11. Please re-install the extension and all should be good
